syntax="proto3";

package user_service;

service UserService {
    rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse) {}
    rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse) {}
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
    rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {}
}

message GetUserByIdRequest {
    int32 id = 1;
}

message GetUserByIdResponse {
    int32 id = 1;
    string email = 2;
    string hashed_password = 3;
    string refresh_token = 4;
}

message DeleteUserRequest {
    int32 id = 1;
}

message DeleteUserResponse {
    bool success = 1;
}

message CreateUserRequest {
    string email = 1;
    string hashed_password = 2;
    string refresh_token = 3;
}

message CreateUserResponse {
    int32 id = 1;
    string email = 2;
}

message UpdateUserRequest {
    int32 id = 1;
    optional string hashed_password = 2;
    optional string refresh_token = 3;
}

message UpdateUserResponse {
    int32 id = 1;
    string email = 2;
}